src/lib/utils/createSendEventForFacet.ts

```diff
@@ -0,0 +1,50 @@
+import { AlgoliaSearchHelper } from 'algoliasearch-helper';
+import { InstantSearch } from '../../types';
+import isFacetRefined from './isFacetRefined';
+
+type BuiltInSendEventForFacet = (eventType: string, facetValue: string) => void;
+type CustomSendEventForFacet = (customPayload: any) => void;
+
+type SendEventForFacet = BuiltInSendEventForFacet & CustomSendEventForFacet;
+
+export default function createSendEventForFacet({
+  instantSearchInstance,
+  helper,
+  attribute,
+  widgetType,
+}: {
+  instantSearchInstance: InstantSearch;
+  helper: AlgoliaSearchHelper;
+  attribute: string;
+  widgetType: string;
+}): SendEventForFacet {
+  const sendEventForFacet: SendEventForFacet = (...args) => {
+    const [eventType, facetValue, eventName = 'Filter Applied'] = args;
+    if (eventType === 'click' && (args.length === 2 || args.length === 3)) {
+      if (!isFacetRefined(helper, attribute, facetValue)) {
+        // send event only when the facet is being checked "ON"
+        instantSearchInstance.sendEventToInsights({
+          insightsMethod: 'clickedFilters',
+          widgetType,
+          eventType,
+          payload: {
+            eventName,
+            index: helper.getIndex(),
+            filters: [`${attribute}:${JSON.stringify(facetValue)}`],
```

Comment:
I stringified `facetValue` because the value could contain special characters like double qoutes.
Is it enough? I think we should have similar logic somewhere in either InstantSearch, helper or client.
I haven't looked into them yet.
If someone already knows, can you point me to it?

Reply:
It's not for the js client. The [doc](https://www.algolia.com/doc/api-reference/api-methods/clicked-filters/#method-param-filters) refers to `filters` in the api reference. So I thought I should follow the `filters` format which requires the value to be quoted when it's with a space or quotes, etc.